common/src/main/java/com/tc/net/protocol/transport/ConnectionHealthCheckerImpl.java

```diff
@@ -91,10 +99,9 @@ public void start() {
   }
 
   @Override
-  public void stop() {
-    if (shutdown.attemptSet()) {
-      monitorThreadEngine.stop();
-      monitorThread.cancel();
+  public synchronized void stop() {
+    if (task != null && !task.isCancelled()) {
+      task.cancel(true);
 // don't bother to join the monitorThread here.  shutdown should take care of all the
```

Comment:
is this comment relevant ? the scheduled thread is not in the TCThreadGroup.

Reply:
yeah.  probably not relevant any longer.